torch/_higher_order_ops/invoke_subgraph.py

```diff
@@ -211,8 +211,11 @@ def invoke_subgraph_func(ctx, subgraph, identifier, operands):
 
 @invoke_subgraph.py_impl(FakeTensorMode)
 def invoke_subgraph_fake_tensor_mode(mode, subgraph, identifier, operands):
-    # TODO(anijain2305) - Implement fake tensor caching.
-    return subgraph(*operands)
+    outputs = mode.has_traced_invoke_subgraph_before(identifier)
+    if outputs is None:
+        outputs = subgraph(*operands)
+        mode.add_traced_invoke_subgraph(identifier, outputs)
+    return outputs
```

Comment:
Why can't we re-use the existing FakeTensor cache here?

Reply:
Also, we don't want to exactly return the outputs, I think we want to return a deep-copied version of the outputs. FakeTensor's cache handles this

Reply:
I was planning to do something like 

```
return mode.__torch_dispatch(invoke_subgraph,[], (subgraph, identifier, operands))
```

However, I realized that existing fake tensor cache would not be able to hash subgraph.

Reply:
How difficult is it to update the existing FakeTensorCache to hash the subgraph? In general we'll want FakeTensorCache to cache HOPs